app/ante/fee_checker.go

```diff
@@ -11,41 +16,38 @@ const (
 	priorityScalingFactor = 1_000_000
 )
 
-// checkTxFeeWithValidatorMinGasPrices implements the default fee logic, where the minimum price per
-// unit of gas is fixed and set by each validator, and the tx priority is computed from the gas price.
-func checkTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
+// CheckTxFeeWithGlobalMinGasPrices implements the default fee logic, where the minimum price per
+// unit of gas is fixed and set globally, and the tx priority is computed from the gas price.
+func CheckTxFeeWithGlobalMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error) {
 	feeTx, ok := tx.(sdk.FeeTx)
 	if !ok {
 		return nil, 0, errors.Wrap(sdkerror.ErrTxDecode, "Tx must be a FeeTx")
 	}
 
-	feeCoins := feeTx.GetFee()
+	fee := feeTx.GetFee().AmountOf(appconsts.BondDenom)
 	gas := feeTx.GetGas()
 
-	// Ensure that the provided fees meet a minimum threshold for the validator,
-	// if this is a CheckTx. This is only for local mempool purposes, and thus
-	// is only ran on check tx.
-	if ctx.IsCheckTx() {
-		minGasPrices := ctx.MinGasPrices()
-		if !minGasPrices.IsZero() {
-			requiredFees := make(sdk.Coins, len(minGasPrices))
+	// global minimum fee only applies to app versions greater than one
+	if ctx.BlockHeader().Version.App > v1.Version {
+		// check for overflow, gas is now capped at MaxInt64
+		if gas > math.MaxInt64 {
+			return nil, 0, errors.Wrap(sdkerror.ErrInvalidGasLimit, "overflow: gas exceeds MaxInt64")
+		}
 
-			// Determine the required fees by multiplying each required minimum gas
-			// price by the gas limit, where fee = ceil(minGasPrice * gasLimit).
-			glDec := sdk.NewDec(int64(gas))
-			for i, gp := range minGasPrices {
-				fee := gp.Amount.Mul(glDec)
-				requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
-			}
+		// convert the global minimum gas price to a big.Int
+		globalMinGasPrice, err := sdk.NewDecFromStr(fmt.Sprintf("%f", appconsts.GlobalMinGasPrice))
+		if err != nil {
+			return nil, 0, errors.Wrap(err, "invalid GlobalMinGasPrice")
+		}
 
-			if !feeCoins.IsAnyGTE(requiredFees) {
-				return nil, 0, errors.Wrapf(sdkerror.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins, requiredFees)
-			}
+		minFee := globalMinGasPrice.MulInt64(int64(gas)).TruncateInt()
+		if !fee.GTE(minFee) {
```

Comment:
Because we are truncating this will need to be `!fee.GT(minFee)`

Reply:
I think it's unintuitive to replace gte with gt since we should allow fee to be equal to. minFee. I think the only time we'd need gt is if minFee truncated to 0. but I added a rule for that edge case.